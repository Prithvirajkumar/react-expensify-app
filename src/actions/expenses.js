import uuid from "uuid";

export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0,
} = {}) => {
  return {
    type: "ADD_EXPENSE",
    expense: {
      id: uuid(),
      description: description,
      note: note,
      amount: amount,
      createdAt: createdAt,
    },
  };
};

export const removeExpense = ({ id } = {}) => {
  return {
    type: "REMOVE_EXPENSE",
    id,
  };
};

export const editExpense = (id, updates) => {
  return {
    type: "EDIT_EXPENSE",
    id,
    updates,
  };
};
