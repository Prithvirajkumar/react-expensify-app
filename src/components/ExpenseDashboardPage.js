import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

export const ExpenseDashboardPage = () => {
  return (
    <div>
      <ExpenseListFilters></ExpenseListFilters>
      <ExpenseList></ExpenseList>
    </div>
  );
};

export default ExpenseDashboardPage;
