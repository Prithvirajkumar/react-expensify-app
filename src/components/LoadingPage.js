import React from "react";

const LoadingPage = () => (
  <div className="loader">
    <img className="loader__image" src="/images/loader.gif"></img>
  </div>
);

export default LoadingPage;
