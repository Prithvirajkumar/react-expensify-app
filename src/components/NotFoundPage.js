import React from "react";
import { Link } from "react-router-dom";

export const NotFoundPage = () => {
  return (
    <div>
      404 <Link to="/">GO Home</Link>
    </div>
  );
};

export default NotFoundPage;
