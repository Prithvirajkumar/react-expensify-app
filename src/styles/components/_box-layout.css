.box-layout {
  align-items: center;
  background-image: url("/images/bg.jpg");
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: rgba(255, 255, 255, 0.85);
  border-radius: 5px;
  text-align: center;
  width: 25rem;
  padding: var(--l-size) var(--m-size);
}

.box-layout__title {
  margin: 0 0 var(--m-size) 0;
  line-height: 1;
}
