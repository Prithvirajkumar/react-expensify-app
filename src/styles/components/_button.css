.button {
  background: var(--blue);
  color: var(--white);
  display: inline-block;
  border: none;
  font-weight: 300;
  font-size: var(--font-size-large);
  padding: var(--s-size);
  border-radius: 3px;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background-color: #888888;
}
