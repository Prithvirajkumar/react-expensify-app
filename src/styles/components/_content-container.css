.content-container {
  margin: 0 auto;
  padding: 0 var(--m-size);
  max-width: 80rem;
}
