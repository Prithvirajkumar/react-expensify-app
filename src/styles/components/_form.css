.form {
  display: flex;
  flex-direction: column;
}

.form > * {
  margin-bottom: var(--m-size);
}

.form__error {
  margin: 0 0 var(--m-size) o;
  font-style: italic;
}
