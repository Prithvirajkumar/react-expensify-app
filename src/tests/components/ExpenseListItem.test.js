import React from "react";
import { shallow } from "enzyme";
import { ExpenseListItem } from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";
import toJSON from "enzyme-to-json";

test("Should render ExpenseListItem with expense", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[1]}></ExpenseListItem>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
