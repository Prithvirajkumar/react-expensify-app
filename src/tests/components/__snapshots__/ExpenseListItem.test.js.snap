// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem with expense 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    109500
     - 
    -345600000
  </p>
</div>
`;
